tt_metal/detail/util.hpp

```diff
@@ -24,10 +25,10 @@ namespace tt::tt_metal::detail{
         return tt::tile_size(data_format);
     }
 
-    inline uint32_t SizeBytesPerBank(uint32_t size_bytes, uint32_t page_size_bytes, uint32_t num_banks, uint32_t alignment_bytes) {
+    inline DeviceAddr SizeBytesPerBank(DeviceAddr size_bytes, DeviceAddr page_size_bytes, uint32_t num_banks, uint32_t alignment_bytes) {
         TT_ASSERT(page_size_bytes > 0 and size_bytes % page_size_bytes == 0, "Page size {} should be divisible by buffer size {}", page_size_bytes, size_bytes);
```

Comment:
Should this be size_t instead?  I'm not following why we use DeviceAddr

Reply:
size_t would limit us to uint32_t on a 32 bit system. Wormhole, for example, has a 64 bit address space. Using size_t would tie the device address width to host, which would be incorrect.